### PR TITLE
Draft generate latlon box

### PR DESCRIPTION
**Pull request by hongyuchen1030**
1. Finish the skeleton code of generating latlon box
2. Add some helper functions in helper.py for latlonbox operation

### CODE REVIEW

File: uxarray/helpers.py [hunk 1]
```diff
@@ -89,3 +91,275 @@ def determine_file_type(filepath):
...
+       Exception: Logic Errors
+    """
+    reference_tolerance = 1.0e-12
+    dx = node_coord[0]
+    dy = node_coord[1]
+    dz = node_coord[2]
+
+    d_mag_2 = dx * dx + dy * dy + dz * dz
+
+    if np.absolute(d_mag_2 - 1.0) >= 0.01:
```
Comment (rajeeja) [inline uxarray/helpers.py:341]: sqrt of d_mag_2 needed?

why raise an exception?  rather: just convert to unit magnitude with a warning.

Comment (hongyuchen1030) [reply uxarray/helpers.py:341]: Thank you for your information. I will modify it accordingly in the following commits.

**Comment by rajeeja** [comment]
Please resolve conflicts, I just approved the test runs for this branch.

**Comment by erogluorhan** [comment]
Thanks very much for this! 

I'd like to express an important thing before too long: This is much bigger than a manageable, reviewable PR. It includes more than one logically separate tasks (e.g. validate, Edge class, and lat-lon box calculation itself are tasks that should be implemented and reviewed in different PRs). We should split this into as many PRs as needed and get them reviewed separately.

**Pull request closed without merging**